spec/Checker/OrderCustomerRelationCheckerSpec.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace spec\Sylius\CustomerReorderPlugin\Checker;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\CustomerReorderPlugin\Checker\OrderCustomerRelationCheckerInterface;
+
+final class OrderCustomerRelationCheckerSpec extends ObjectBehavior
+{
+   function it_implements_order_customer_relation_checker_interface(): void
+   {
+       $this->shouldImplement(OrderCustomerRelationCheckerInterface::class);
+   }
+
+   function it_returns_true_when_order_was_placed_by_customer(
+       CustomerInterface $customer,
+       OrderInterface $order
+   ): void {
+        $customer->getId()->willReturn('1');
+
+        $order->getCustomer()->willReturn($customer);
+
+        $this->wasOrderPlacedByCustomer($order, $customer)->shouldReturn(true);
```

Comment:
I think something is wrong in this spec 😄 We should have 3 specifications here:

1. Order customer is the same as passed customer
```php
$orderCustomer()->getId()->willReturn(1);
$customer->getId()->willReturn(1);
```

2. Order customer is different than passed customer
```php
$orderCustomer()->getId()->willReturn(1);
$customer->getId()->willReturn(2);
```

3. Order has no customer assigned
```php
$order->getCustomer()->willReturn(null);
```